src/gpu_sysman.c

```diff
@@ -1259,20 +1262,24 @@ static bool gpu_mems_bw(gpu_device_t *gpu) {
     gpu->membw_count = mem_count;
   }
 
+  char *name = METRIC_PREFIX "memory.bw";
   metric_family_t fam_ratio = {
       .help = "Average memory bandwidth usage ratio (0-1) over query interval",
-      .name = METRIC_PREFIX "memory_bw_ratio",
       .type = METRIC_TYPE_GAUGE,
+      .name = name,
+      .unit = "1",
   };
   metric_family_t fam_rate = {
       .help = "Memory bandwidth usage rate (in bytes per second)",
-      .name = METRIC_PREFIX "memory_bw_bytes_per_second",
       .type = METRIC_TYPE_GAUGE,
+      .name = name,
+      .unit = "By/s",
   };
   metric_family_t fam_counter = {
       .help = "Memory bandwidth usage total (in bytes)",
-      .name = METRIC_PREFIX "memory_bw_bytes_total",
       .type = METRIC_TYPE_COUNTER,
+      .name = name,
+      .unit = "By",
   };
```

Comment:
I think OTEL naming requirements / recommendations are badly thought out and inconsistent (e.g. `system.cpu.frequency` vs. `hw.cpu.speed`).

> Each metric needs a unique name. Here and below.

While for these I could use `.utilization`, `.rate` and `.usage`, and e.g. `throttled.time` is fine for total counter, something like `throttled.utilization` for ratio sounds completely bonkers.

For now I'll use throttling ratio without suffix, but that does not feel quite correct either.  Happily Prometheus plugin adds `_ratio` suffix to it, which at least makes it understandable to users.

Value, & value ratio (against max/limit) for frequency and temperature, are also problematic.  OTEL spec requires value itself not to have suffix, and as they are not consumable, but result of internal state & environmental factors [1], `.utilization` suffix is wrong for their ratios.

Btw. Several OTEL metrics specify just `.limit` metric without a ratio one.  While that is fine with frontends like Grafana / Prometheus, not having those (especially rate) when accessing collectd data directly (for validation & local GPU checks) sucks.  Switching from ratios to reporting limits would also be sad as Sysman plugin already has quite a bit of code for showing user specified mix of values, their rates and ratios for most metrics...

Suggestions?

----

[1] GPU load, cooling, air temperature

PS. OTEL fan metric actually use `_ratio` suffix, see: https://opentelemetry.io/docs/specs/semconv/system/hardware-metrics/#hwfan---fan-metrics (!)

(Sysman frontend provides API for Fan metrics, although backend support for that is still missing.)

Reply:
The conventions for hardware are in the "experimental" status, i.e. now would be the time to propose changes. I'd encourage you to join the OpenTelemetry Slack and start a conversation there: https://slack.cncf.io/

Not every ratio needs to be a "utilization". For example, the "HTTP 5xx errors / total number of requests" ratio makes no sense expressed as "utilization". Likewise, the "overclocked / non-overclocked frequency" ratio is not a "utilization". IMHO `hw.gpu.frequency_ratio` is absolutely fine.

For temperature, I don't find a ratio meaningful. As a user, I'd prefer to get current value + limit metrics.

For throttled, I suggest we report a "throttled.time" metric that is scaled to seconds per second, and remove the ratio metric. I'd prefer to use the floating point counters for this once they're merged, but we can use a gauge for now.

Reply:
Before using `_ratio` names, `write_prometheus` needs to be fixed to avoid duplicating that suffix => added that to list in PR description.

Yes, temperature ratio is probably redundant, as I would imagine GPUs to have quite similar limits.  For other metrics, ratios are nice as one does not need to code per-device rules for problematic situations, and for #4177 use-case a pre-calculated one is _much_ preferred.

Throttling time was in Sysman API from the start, but I've yet to seem backend support for it, so one option would be to ifdef it out, or split it to another branch, until there's backend support for it (frequency metric does have throttled attribute, but that's just a point sample).

PS. Just a reminder...  It will be 1+ week until I'll update this again (or review other PRs).